compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5079,6 +5066,47 @@ public void visit(BLangBinaryExpr binaryExpr, AnalyzerData data) {
         data.resultType = types.checkType(binaryExpr, actualType, data.expType);
     }
 
+    private BType getBinaryOperandType(BLangExpression operand, BLangBinaryExpr binaryExpr, BType referredExpType,
+                                       SymbolEnv env, AnalyzerData data) {
+        if ((referredExpType.tag == TypeTags.FLOAT || referredExpType.tag == TypeTags.DECIMAL ||
+                isOptionalFloatOrDecimal(referredExpType)) && isNumericLiteral(operand)) {
```

Comment:
Can't we use `requireTypeInference` instead of `isNumericLiteral`?

Reply:
Tests are passing with `requireTypeInference`. But in that method, it returns true for other non numeric literal cases. Even if those case does not return true in binary-expr context, I think it is better to have a separate method. WDYT?